.gitlab-ci.yml

```diff
@@ -79,40 +81,66 @@ test-flatbuf:
     script:
         - (cd java/test_flatbuf1 && ./run.sh)
 
+# Template for souffle tests
+.test-imported-souffle:
+    extends: .install-ddlog
+    tags:
+        - ddlog-ci-1
+
 # Tests from the souffle github repo.
 test-imported-souffle-tests1:
-    extends: .install-ddlog
+    extends: .test-imported-souffle
     script:
         - (cd test && ./run-souffle-tests-in-batches.py 0 24)
 
-# Slow
-#test-imported-souffle-tests2:
-#    extends: .install-ddlog
-#    script:
-#        - (cd test && ./run-souffle-tests-in-batches.py 25 49)
+test-imported-souffle-tests2:
+    extends: .test-imported-souffle
+    script:
+        - (cd test && ./run-souffle-tests-in-batches.py 25 49)
 
-# Times out or runs out of memory on gitlab
-#test-imported-souffle-tests3:
-#    extends: .install-ddlog
-#    script:
-#        - (cd test && ./run-souffle-tests-in-batches.py 50 74)
+test-imported-souffle-tests3:
+    extends: .test-imported-souffle
+    script:
+        - (cd test && ./run-souffle-tests-in-batches.py 50 74)
 
 test-imported-souffle-tests4:
-    extends: .install-ddlog
+    extends: .test-imported-souffle
     script:
         - (cd test && ./run-souffle-tests-in-batches.py 75 99)
 
 test-imported-souffle-tests5:
-    extends: .install-ddlog
+    extends: .test-imported-souffle
     script:
         - (cd test && ./run-souffle-tests-in-batches.py 100 124)
 
 test-imported-souffle-tests6:
-    extends: .install-ddlog
+    extends: .test-imported-souffle
     script:
         - (cd test && ./run-souffle-tests-in-batches.py 125 149)
 
 test-imported-souffle-tests7:
-    extends: .install-ddlog
+    extends: .test-imported-souffle
     script:
         - (cd test && ./run-souffle-tests-in-batches.py 150 175)
+
+# Test ovn-northd-ddlog
+test-ovn-northd:
+    extends: .install-ddlog
+    tags:
+        - ddlog-ci-2
+    script:
+        # TODO: use main OVS repo once these changes are there.
+        - (cd /root &&
+          git clone https://github.com/numansiddique/ovs.git &&
+          cd ovs && git checkout ovs_ddlog_patches &&
+          ./boot.sh &&
+          ./configure  &&
+          make -j6)
+        # TODO: maintain and eventually remove the list of tests.
+        # TODO: use -j6 once build script is fixed
+        - (cd /root &&
+          git clone https://github.com/ovn-org/ovn.git &&
+          cd ovn && git checkout ddlog-dev-v2 &&
+          ./boot.sh &&
+          ./configure --with-ddlog=/root/.local/ddlog/lib --with-ovs-source=/root/ovs &&
+          make check -j1 TESTSUITEFLAGS="117-135 138-141 143-145 147-149 151-152 154-161 167-169 172 174-181 183-185 187-189 191 193-196 198-199 201-202")
```

Comment:
this looks very brittle, but I don't know if there is any alternative

Reply:
It is. I plan to maintain this list until it becomes empty.